### PR TITLE
OBSDOCS-215: how-to-enable-dedicated-service-monitor

### PR DESCRIPTION
**Pull request by bburt-rh**
Summary: Adds content about dedicated service monitor config

- Aligned team: DevTools
- For branches: OCP 4.10+
- Jira: https://issues.redhat.com/browse/OBSDOCS-215
- Direct link to doc preview: https://61893--docspreview.netlify.app/openshift-enterprise/latest/monitoring/configuring-the-monitoring-stack.html#configuring-a-dedicated-service-monitor_configuring-the-monitoring-stack
- SME review: @jan--f 
- QE review: @juzhao 
- Peer review: @shipsing

### CODE REVIEW

**Comment by juzhao** [comment]
/lgtm
waiting for others to review

**Comment by jan--f** [comment]
While all the info is correct, the motivation for this feature could be clearer. https://issues.redhat.com/browse/MON-1949 i the original issue for reference.
By default we collect the metrics that are used for the "[Resource Metrics API](https://kubernetes.io/docs/tasks/debug/debug-cluster/resource-metrics-pipeline/)" in a way that improves alerting accuracy, but produces a timeseries that is not 100% consistent with the real world. In configurations, where Horizontal Pod Autoscaling is required to react very quickly, this can have a negative impact.
When this new feature is enabled we deploy a second ServiceMonitor, that collects additional accurate time series and thus allows for very reactive HPA setups. These time series can however not be used for our alerts. Since we don't want to give up alerting accuracy, the trade-off is to collect 2 additional time series per node

Proposal:

> = Configuring a dedicated service monitors
> 
> You can configure {product-title} core platform monitoring to use dedicated service monitors to collect metrics for the Resource Metrics Pipeline.
> 
> The dedicated service monitors exposes 2 additional metrics from the kublet endpoint and sets the value of the `honorTimestamps` field to `true`.
> 
> By enabling a dedicated service monitor, you can improve the consistency of Prometheus Adapter-based CPU usage measurements used by, for example, the `oc adm top pod` command or the Horizontal Pod Autoscaler.

The instructions on how to enabled this look good though.

**Comment by bburt-rh** [comment]
@jan--f @juzhao - I made the changes suggested by Jan. PTAL and approve if it looks good to you.

File: modules/monitoring-enabling-a-dedicated-service-monitor.adoc [hunk 1]
```diff
@@ -0,0 +1,50 @@
+// Module included in the following assemblies:
+//
+// * monitoring/configuring-the-monitoring-stack.adoc
+
+:_content-type: PROCEDURE
+[id="enabling-a-dedicated-service-monitor_{context}"]
+= Enabling a dedicated service monitor
+
+You can enable core platform monitoring to use a dedicated service monitor by configuring the `dedicatedServiceMonitors` key in the `cluster-monitoring-config` `ConfigMap` object in the `openshift-monitoring` namespace.
```
Comment (shipsing) [inline modules/monitoring-enabling-a-dedicated-service-monitor.adoc:9]:
```suggestion
You can enable a dedicated service monitor by configuring the `dedicatedServiceMonitors` key in the `cluster-monitoring-config` `ConfigMap` object in the `openshift-monitoring` namespace.
```
Are we enabling core platform monitoring or a dedicated service monitor?

File: modules/monitoring-enabling-a-dedicated-service-monitor.adoc [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+metadata:
+  name: cluster-monitoring-config
+  namespace: openshift-monitoring
+data:
+  config.yaml: |
+    k8sPrometheusAdapter:
+      dedicatedServiceMonitors:
+        enabled: true <1>
+----
+<1> Set the `enabled` field to `true` to deploy a dedicated service monitor that exposes the kubelet `/metrics/resource` endpoint.
```
Comment (shipsing) [inline modules/monitoring-enabling-a-dedicated-service-monitor.adoc:41]:
```suggestion
<1> Set the value of the `enabled` field to `true` to deploy a dedicated service monitor that exposes the kubelet `/metrics/resource` endpoint.
```

File: modules/monitoring-enabling-a-dedicated-service-monitor.adoc [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+
+. Save the file to apply the changes automatically.
++
+[WARNING]
+====
+When you save changes to a `cluster-monitoring-config` config map, the pods and other resources in the `openshift-monitoring` project might be redeployed.
```
Comment (shipsing) [inline modules/monitoring-enabling-a-dedicated-service-monitor.adoc:47]:
```suggestion
When you save changes to a `cluster-monitoring-config` config map, the pods and other resources in the `openshift-monitoring` namespace might be redeployed.
```

File: modules/monitoring-enabling-a-dedicated-service-monitor.adoc [hunk 4]
```diff
@@ -0,0 +1,50 @@
...
+The running monitoring processes in that project might also restart.
```
Comment (shipsing) [inline modules/monitoring-enabling-a-dedicated-service-monitor.adoc:48]:
```suggestion
The running monitoring processes in that namespace might also restart.
```

**Review by shipsing — CHANGES_REQUESTED** [review]

File: modules/monitoring-configuring-dedicated-service-monitors.adoc [hunk 5]
```diff
@@ -0,0 +1,13 @@
+// Module included in the following assemblies:
+//
+// * monitoring/configuring-the-monitoring-stack.adoc
+
+:_content-type: CONCEPT
+[id="configuring-a-dedicated-service-monitor_{context}"]
+= Configuring a dedicated service monitor
+
+You can configure {product-title} core platform monitoring to use dedicated service monitors to collect metrics for the resource metrics pipeline.
```
[the remark this answers is not in the document]
Comment (bburt-rh) [reply modules/monitoring-configuring-dedicated-service-monitors.adoc:9]: Thanks, @shipsing. First, monitoring is not a product. It's an OCP feature consisting of a number of components. Within monitoring, there are two component stacks: core platform monitoring (the components installed by default when you install OCP) and user workload monitoring. We can assume that anyone using the feature should know the difference. Thus, I don't see a need for a change here.

File: modules/monitoring-enabling-a-dedicated-service-monitor.adoc (hunk 1 above)
Comment (bburt-rh) [reply modules/monitoring-enabling-a-dedicated-service-monitor.adoc:9]: Good catch. The word `enable` is confusing here. Changing this to `You can configure core platform monitoring to use`

File: modules/monitoring-enabling-a-dedicated-service-monitor.adoc (hunk 2 above)
Comment (bburt-rh) [reply modules/monitoring-enabling-a-dedicated-service-monitor.adoc:41]: Changed per your suggestion.

File: modules/monitoring-enabling-a-dedicated-service-monitor.adoc (hunk 3 above)
Comment (bburt-rh) [reply modules/monitoring-enabling-a-dedicated-service-monitor.adoc:47]: A `namespace` is the Kubernetes word for an OpenShift `project`. They mean the same thing, so I am leaving this as-is

File: modules/monitoring-enabling-a-dedicated-service-monitor.adoc (hunk 4 above)
Comment (bburt-rh) [reply modules/monitoring-enabling-a-dedicated-service-monitor.adoc:48]: A `namespace` is the Kubernetes word for an OpenShift `project`. They mean the same thing, so I am leaving this as-is

**Comment by shipsing** [comment]
Thank you, @bburt-rh. 
LGTM

**Review by shipsing — APPROVED** [review]

**Pull request merged**